lib/datadog/tracing/contrib/faraday/middleware.rb

```diff
@@ -24,7 +24,9 @@ def call(env)
             # Do this once to reduce expensive regex calls.
             request_options = build_request_options!(env)
 
-            Tracing.trace(Ext::SPAN_REQUEST) do |span, trace|
+            on_error = datadog_configuration[:error_handler]
```

Comment:
Did you mean to use 
```suggestion
            on_error = datadog_configuration[:on_error]
```
here instead?

Reply:
Bumping this @marcotc !

Reply:
> Reason being:
> 
>     * `error_handler` is a faraday configuration option that has a different interface:
> 
> 
> > A Proc that accepts a response parameter. If it evaluates to a truthy value, the trace span is marked as an error. By default only sets 5XX responses as errors.
> > [From docs](https://docs.datadoghq.com/tracing/trace_collection/dd_libraries/ruby/#faraday)
> 
>     * `on_error` is a new configuration API for the faraday integration. It's interface is a proc that accepts the span and the error, as required by `Tracing.trace`.
> 
> 
> These two error handlers catch different categories of errors.

This sounds correct to me, which comes back to my original suggestion:
`on_error = datadog_configuration[:error_handler]` is "A `Proc` that accepts a `response` parameter. If it evaluates to a *truthy* value, the trace span is marked as an error. By default only sets 5XX responses as errors." but we are passing it to `Tracing.trace(Ext::SPAN_REQUEST, on_error: on_error)`, which expects a proc that receives a span and an error object.
This means that the configured `datadog_configuration[:error_handler]` proc, which likely looks like `->(response){...}` will actually receive `proc.call(span,error)` and will most likely fail.

Reply:
Oh I see... that's my bad 🤦🏼

Reply:
Not sure why my test passed though 🤔

Reply:
@marcotc Could you help me with something?

- I want to add a test that ensures when there is a no-op `on_error` and we get a `Faraday::ConnectionFailed` but an `error_handler` that always returns `true`, that no error is registered (since this is an error with the `faraday` client rather than with the endpoint being called itself
- However, I'm having trouble getting my tests to run as expected
- I add a `raise` in `call` of `lib/datadog/tracing/contrib/faraday/middleware.rb` and run `bundle exec rake spec:all` and the raise is never hit – I'm not sure why the test is not running
- `bundle exec rspec path/to/file_spec.rb`, the canonical way to run rspec tests, does not work in this repo
- There appears to be no way to run a specific test, based on the `Rakefile`

Can you let me know if I'm missing something? Are these tests not actually running? Is there no way to run individual tests within this repo? Would be great to update the contributor docs to make this a bit clearer since the test configuration is non-standard (i.e. cannot run `bundle exec rspec` successfully).

Thanks for your help!

Reply:
@marcotc Wanted to bump this!

Reply:
@marcotc bump!